servers/rendering/renderer_rd/storage_rd/material_storage.cpp

```diff
@@ -566,6 +566,28 @@ void MaterialStorage::ShaderData::set_default_texture_parameter(const StringName
 Variant MaterialStorage::ShaderData::get_default_parameter(const StringName &p_parameter) const {
 	if (uniforms.has(p_parameter)) {
 		ShaderLanguage::ShaderNode::Uniform uniform = uniforms[p_parameter];
+
+		if (uniform.type == ShaderLanguage::TYPE_STRUCT) {
+			if (uniform.array_size > 0) {
+				Array array;
+
+				for (int i = 0; i < uniform.array_size; i++) {
+					Dictionary dict;
+					for (const ShaderLanguage::ShaderNode::Uniform::Member &member : uniform.members) {
+						dict[member.name] = ShaderLanguage::get_default_datatype_value(member.type, member.array_size, ShaderLanguage::ShaderNode::Uniform::HINT_NONE);
```

Comment:
What happens if the user has an initialization for the struct?

Like for example, having a default for the struct
```glsl
struct A {
    vec3 col;
    float alph;
};
uniform A struct_uniform = A(vec3(1.0),1.0);
```
Wouldn't this just skip over it?

Reply:
It's currently not implemented as stated in the topic header.